enhancements/network/ovn-kubernetes-ipsec.md

```diff
@@ -0,0 +1,298 @@
+---
+title: ovn-kubernetes-ipsec
+authors:
+  - "@mdgray"
+reviewers:
+  - TBD
+  - "@dcbw"
+approvers:
+  - TBD
+creation-date: 2020-09-08
+last-updated: 2020-09-80
+status: provisional
+see-also:
+  - "/enhancements/20190919-OVN-Kubernetes.md"
+replaces:
+  - N/A
+superseded-by:
+  - N/A
+---
+
+# OVN Kubernetes IPsec Enablement
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The OVN Kubernetes network plugin uses OVN (https://www.ovn.org) to instantiate
+virtual networks for Kubernetes. These virtual networks use Geneve, a network
+virtualization overlay encapsulation protocol, to tunnel traffic across the
+underlay network between Kubernetes Nodes.
+
+IPsec is a protocol suite that enables secure network communications between
+IP endpoints by providing the following services:
+- Confidentiality
+- Authentication
+- Data Integrity
+
+OVN tunnel traffic is transported by physical routers and switches. These
+physical devices could be untrusted (devices in public network) or might be
+compromised. Enabling IPsec encryption for this tunnel traffic can prevent the
+traffic data from being monitored and manipulated.
+
+This enhancement allows cluster operators to enable traffic to be encapsulated
+by the IPsec protocol between worker nodes.
+
+## Motivation
+
+Encryption services are recommended when traffic is traversing an untrusted
+network. Encryption services may also be required for regulatory or compliance
+reasons.
+
+### Goals
+
+- Provide an option to enable IPsec encryption of all inter-worker node
+tunnel traffic across an entire Kubernetes cluster that has been configured with
+OVN Kubernetes.
+  - Assume that all nodes will be provisioned with certificates that have been
+  signed by the OpenShift Container Platform internal certificate authority
+  (CA).
+  - This configuration option will only be available on Geneve tunnels
+- All components used to implement this must be FIPs-compliant cryptographic
+components.
+- Provide an implementation that will be deployable on both Public and Private
+Cloud infrastructure. This will require that all inter-worker node traffic can
+be routed by the underlay network without reconfiguration of underlay firewall
+policies.
+- Reuse the same certificates, keys and CA used by the Cluster Management TLS
+implementation.
+
+### Non-Goals
+
+- Will not provide any Hardware Offload of the IPsec protocol. Although some
+checksums may be offloaded by standard NICs without any special enablement.
```

Comment:
Doesn't IPSec hardware offload normally happen via automatic detection of kernel and NIC capabilites?  By my read of ipset.conf(5), the nic-offload setting defaults to "auto", so unless we're explicitly opting out of this it should just happen wherever it can.

Reply:
@lack Yes you are right. I was trying to state that in the comment but I will make it clearer.

Reply:
I will remove the section. No need to reference this.